filestore/protoboards.go

```diff
@@ -0,0 +1,118 @@
+package filestore
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/influxdata/chronograf"
+)
+
+// ProtoboardExt is the the file extension searched for in the directory for protoboard files
+const ProtoboardExt = ".json"
+
+// Protoboards are instantiable JSON representation of dashbards.  Implements ProtoboardsStore.
+type Protoboards struct {
+	Dir     string                                      // Dir is the directory containing protoboard json definitions
+	Load    func(string) (chronograf.Protoboard, error) // Load receives filename, returns a Protoboard from json file
+	ReadDir func(dirname string) ([]os.FileInfo, error) // ReadDir reads the directory named by dirname and returns a list of directory entries sorted by filename.
+	IDs     chronograf.ID                               // ID generates unique ids for new protoboards
+	Logger  chronograf.Logger
+}
+
+// NewProtoboards constructs a protoboard store wrapping a file system directory
+func NewProtoboards(dir string, ids chronograf.ID, logger chronograf.Logger) chronograf.ProtoboardsStore {
+	return &Protoboards{
+		Dir:     dir,
+		Load:    protoboardLoadFile,
+		ReadDir: ioutil.ReadDir,
+		IDs:     ids,
+		Logger:  logger,
+	}
+}
+
+func protoboardLoadFile(name string) (chronograf.Protoboard, error) {
+	octets, err := ioutil.ReadFile(name)
+	if err != nil {
+		return chronograf.Protoboard{}, chronograf.ErrProtoboardNotFound
+	}
+	var protoboard chronograf.Protoboard
+	if err = json.Unmarshal(octets, &protoboard); err != nil {
+		return chronograf.Protoboard{}, chronograf.ErrProtoboardInvalid
+	}
+	return protoboard, nil
+}
+
+// All returns all protoboards from the directory
+func (a *Protoboards) All(ctx context.Context) ([]chronograf.Protoboard, error) {
+	files, err := a.ReadDir(a.Dir)
+	if err != nil {
+		return nil, err
+	}
+
+	protoboards := []chronograf.Protoboard{}
+	for _, file := range files {
+		if path.Ext(file.Name()) != ProtoboardExt {
+			continue
+		}
+		if protoboard, err := a.Load(path.Join(a.Dir, file.Name())); err != nil {
+			continue // We want to load all files we can.
+		} else {
+			protoboards = append(protoboards, protoboard)
+		}
+	}
+
+	return protoboards, nil
+}
+
+// Get returns a protoboard file from the protoboard directory
+func (a *Protoboards) Get(ctx context.Context, ID string) (chronograf.Protoboard, error) {
+	l, file, err := a.idToFile(ID)
+	if err != nil {
+		return chronograf.Protoboard{}, err
+	}
+
+	if err != nil {
+		if err == chronograf.ErrProtoboardNotFound {
```

Comment:
Why not just alway log the error here?

Reply:
copied from layouts. want us to just pass through whatever error?